### DISCUSSION TITLE
Arduino Nano IOT 33

### DISCUSSION THREAD

**Discussion by Bipman1** [lathoub/Arduino-BLE-MIDI#75] [Q&A] [answered]
Hi

Should this library work with the Nano IOT 33?  I've installed the correct libraries and am using the following as a test 

#include<MIDI.h>
#include <ArduinoBLE.h>
#include <BLEMIDI_Transport.h>
#include <hardware/BLEMIDI_ArduinoBLE.h>

BLEMIDI_CREATE_DEFAULT_INSTANCE()

void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}>

But it doesn't compile due to errors below

`n file included from C:\Users\steph\AppData\Local\Temp\.arduinoIDE-unsaved2023112-1980-1842e3e.n3lbk\sketch_feb12a\sketch_feb12a.ino:3:0:
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/hardware/BLEMIDI_ArduinoBLE.h:5:24: error: 'SERVICE_UUID' was not declared in this scope
 BLEService midiService(SERVICE_UUID);
                        ^~~~~~~~~~~~
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/hardware/BLEMIDI_ArduinoBLE.h:5:24: note: suggested alternative:
In file included from C:\Users\steph\AppData\Local\Temp\.arduinoIDE-unsaved2023112-1980-1842e3e.n3lbk\sketch_feb12a\sketch_feb12a.ino:2:0:
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/BLEMIDI_Transport.h:21:26: note:   'bleMidi::SERVICE_UUID'
 static const char *const SERVICE_UUID        = "03b80e5a-ede8-4b33-a751-6ce34ec4c700";
                          ^~~~~~~~~~~~
In file included from C:\Users\steph\AppData\Local\Temp\.arduinoIDE-unsaved2023112-1980-1842e3e.n3lbk\sketch_feb12a\sketch_feb12a.ino:3:0:
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/hardware/BLEMIDI_ArduinoBLE.h:6:34: error: 'CHARACTERISTIC_UUID' was not declared in this scope
 BLEStringCharacteristic midiChar(CHARACTERISTIC_UUID,  // standard 16-bit characteristic UUID
                                  ^~~~~~~~~~~~~~~~~~~
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/hardware/BLEMIDI_ArduinoBLE.h:6:34: note: suggested alternative:
In file included from C:\Users\steph\AppData\Local\Temp\.arduinoIDE-unsaved2023112-1980-1842e3e.n3lbk\sketch_feb12a\sketch_feb12a.ino:2:0:
d:\Users\steph\Arduino\libraries\Arduino-BLE-MIDI-master\src/BLEMIDI_Transport.h:22:26: note:   'bleMidi::CHARACTERISTIC_UUID'
 static const char *const CHARACTERISTIC_UUID = "7772e5db-3868-4112-a1a9-f2669d106bf3";
                          ^~~~~~~~~~~~~~~~~~~

exit status 1

Compilation error: exit status 1`

Any idea ? (I used the 'code' selection but it doesn't seem to have worked)

Bipman

**Comment by Bipman1** [comment]
Version 2.1.0 works OK.

**Comment by lathoub** [comment, marked as answer]
Please use this branch: https://github.com/lathoub/Arduino-BLE-MIDI/tree/Arduino-Nano-BLE-33

And report back if all work well (I have asked others before, never got feedback)

**Comment by Bipman1** [comment]
Hi

That's great thanks.


On Sun, 12 Feb 2023 at 21:30, lathoub ***@***.***> wrote:

> Please use this branch:
> https://github.com/lathoub/Arduino-BLE-MIDI/tree/Arduino-Nano-BLE-33
>
> And report back if all work well (I have asked others before, never got
> feedback)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lathoub/Arduino-BLE-MIDI/discussions/75#discussioncomment-4951383>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY2WDJH6FF6XLRR7JWBCOL3WXFI7BANCNFSM6AAAAAAUZQHXZI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 

Regards


Stephen Gray



Check out the links below for Skynet ambient music

YouTube
<https://www.youtube.com/channel/UC4cOaAR33QRWzmYxr44Uu-Q?view_as=subscriber>
https://SoundCloud.com\Stephen-Gray-260393121

**Comment by LamontOmlanduH9** [comment]
It seems that the error is due to the missing declaration of SERVICE_UUID and CHARACTERISTIC_UUID in the BLEMIDI_ArduinoBLE.h header file, and the suggested alternatives are present in the BLEMIDI_Transport.h header file, so you may need to check if the correct libraries have been installed and if the headers are properly included. As for compatibility with Nano IOT 33, it should work since it is compatible with the ArduinoBLE library.

**Comment by Bipman1** [comment]
Thanks!

**Reply by matpowel** [reply]
2025 and getting this exact issue, the branch seems to be deleted. What's the solution? Do we just need to fix the const var references to scope them properly?

**Reply by matpowel** [reply]
For future travelers, this was the fix (just adding the missing namespacing on the UUID vars, BLEMIDI_NAMESPACE::).

```
--- a/src/hardware/BLEMIDI_ArduinoBLE.h
+++ b/src/hardware/BLEMIDI_ArduinoBLE.h
@@ -2,8 +2,8 @@
 
 #include <ArduinoBLE.h>
 
-BLEService midiService(SERVICE_UUID);
-BLEStringCharacteristic midiChar(CHARACTERISTIC_UUID,  // standard 16-bit characteristic UUID
+BLEService midiService(BLEMIDI_NAMESPACE::SERVICE_UUID);
+BLEStringCharacteristic midiChar(BLEMIDI_NAMESPACE::CHARACTERISTIC_UUID,  // standard 16-bit characteristic UUID
     BLERead | BLEWrite | BLENotify | BLEWriteWithoutResponse, 16); // remote clients will be able to get notifications if this characteristic changes
 
 #define BLE_POLLING
```
